xbmc/cores/VideoPlayer/VideoRenderers/WinRenderBuffer.cpp

```diff
@@ -0,0 +1,647 @@
+/*
+ *      Copyright (C) 2005-2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <ppl.h>
+#include <ppltasks.h>
+
+#include "WinRenderBuffer.h"
+#include "DVDCodecs/DVDCodecUtils.h"
+#include "utils/log.h"
+#include "utils/win32/gpu_memcpy_sse4.h"
+#include "utils/CPUInfo.h"
+#include "utils/win32/memcpy_sse2.h"
+#include "windowing/windows/WinSystemWin32DX.h"
+#include "WinRenderer.h"
+
+#define PLANE_Y 0
+#define PLANE_U 1
+#define PLANE_V 2
+#define PLANE_UV 1
+#define PLANE_D3D11 0
+
+CRenderBuffer::CRenderBuffer()
+  : loaded(false)
+  , frameIdx(0)
```

Comment:
ctor initialization is hard to maintain. much better is c++11 syle by assignment in header
your code, your decision